Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/core/impl/ConceptListService.java

```diff
@@ -0,0 +1,78 @@
+package edu.asu.conceptpower.app.core.impl;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.conceptpower.app.core.IConceptListService;
+import edu.asu.conceptpower.app.core.model.impl.ConceptList;
+import edu.asu.conceptpower.app.core.repository.IConceptListRepository;
+
+/*
+ * 
+ * @author Keerthivasan Krishnamurthy
+ * 
+ * */
+@Service
+public class ConceptListService implements IConceptListService{
+    
+    @Autowired
+    private IConceptListRepository conceptListRepository;
+    
+
+    protected final String LIST_PREFIX = "LIST";
+    
+    @Override
+    public void addConceptList(String name, String description) {
+        ConceptList conceptList = new ConceptList();
+        conceptList.setConceptListName(name);
+        conceptList.setDescription(description);
+        conceptList.setId(generateId(LIST_PREFIX));
+        conceptListRepository.save(conceptList);
+    }
+
+    @Override
+    public void deleteConceptList(String name) {
+        // TODO: Must check whether should we proceed with delete by conceptList name
+        conceptListRepository.deleteByConceptListName(name);
+    }
+
+    @Override
+    public List<ConceptList> getAllConceptLists() {
+        return conceptListRepository.findAll();
+    }
+
+    @Override
+    public ConceptList getConceptList(String name) {
+        // TODO: Must check whether should we proceed with delete by conceptList name
+        return conceptListRepository.findByConceptListName(name);
+    }
+
+    @Override
+    public void storeModifiedConceptList(ConceptList list, String listname) {
+        // TODO Must see if we need listname argument
+        conceptListRepository.save(list);
+    }
+
+    @Override
+    public boolean checkExistingConceptList(String name) {
+        return conceptListRepository.findByConceptListName(name) != null;
+    }
+    
+    protected String generateId(String prefix) {
```

Comment:
actually, let's switch to database generated ids for lists at least. If you check one of the other projects (citesphere I believe has it for example, or virtual spaces), there is id generated that uses a prefix and then uses a database counter to generate ids.